runtime/src/bank.rs

```diff
@@ -4808,11 +4808,24 @@ impl Bank {
     }
 
     fn apply_spl_token_v2_self_transfer_fix(&mut self) {
-        if let Some(mut account) = self.get_account(&inline_spl_token_v2_0::id()) {
-            self.capitalization.fetch_sub(account.lamports, Relaxed);
-            account.lamports = 0;
-            self.store_account(&inline_spl_token_v2_0::id(), &account);
-            self.remove_executor(&inline_spl_token_v2_0::id());
+        if let Some(old_account) = self.get_account(&inline_spl_token_v2_0::id()) {
+            if let Some(new_account) =
+                self.get_account(&inline_spl_token_v2_0::new_token_program::id())
+            {
+                // Burn lamports in the old token account
+                self.capitalization.fetch_sub(old_account.lamports, Relaxed);
+
+                // Transfer new token account to old token account
+                self.store_account(&inline_spl_token_v2_0::id(), &new_account);
+
+                // Clear new token account
+                self.store_account(
+                    &inline_spl_token_v2_0::new_token_program::id(),
+                    &Account::default(),
+                );
+
+                self.remove_executor(&inline_spl_token_v2_0::id());
+            }
```

Comment:
NIT: It might pay to add some logging down here?  Should we botch the new address copy-pasta or similar

Reply:
Do you think the yelling in the patch for the test node is enough, assuming the test node uses the tip of the v1.4 branch plus that patch?
```rust
 error!(
                "REDIRECT {} to {}",
                pubkey,
                inline_spl_token_v2_0::new_token_program::id()
            );
```

Reply:
I think @t-nelson is talking about logs for when the activation/old-token-rewrite actually happens, although I suppose that test node will let us know by failing to load the now-empty `new_token_program` account for subsequent token ixs and forking itself off.
A log or datapoint here seems reasonable to me. I like datapoint just to confirm at a glance that all the nodes on the network performed the rewrite.

Reply:
10-4, like this edfff3853bb56f57180db88a13146477957015a0?

Reply:
We'll also be able to tell after the fact by simply inspecting the new token program account, it'll be deallocated

Reply:
> 10-4, like this [edfff38](https://github.com/solana-labs/solana/commit/edfff3853bb56f57180db88a13146477957015a0)?

Yep! That'll work